internal/uvm/plan9.go

```diff
@@ -42,95 +54,76 @@ func (uvm *UtilityVM) AddPlan9(hostPath string, uvmPath string, readOnly bool) e
 	if readOnly {
 		flags |= shareFlagsReadOnly
 	}
+	if restrict {
+		flags |= shareFlagsRestrictFileAccess
+	}
 
 	uvm.m.Lock()
-	defer uvm.m.Unlock()
-	if uvm.plan9Shares == nil {
-		uvm.plan9Shares = make(map[string]*plan9Info)
-	}
-	if _, ok := uvm.plan9Shares[hostPath]; !ok {
-		uvm.plan9Counter++
+	index := uvm.plan9Counter
+	uvm.plan9Counter++
+	uvm.m.Unlock()
+	name := strconv.FormatUint(index, 10)
 
-		modification := &hcsschema.ModifySettingRequest{
-			RequestType: requesttype.Add,
-			Settings: hcsschema.Plan9Share{
-				Name:  fmt.Sprintf("%d", uvm.plan9Counter),
-				Path:  hostPath,
-				Port:  int32(uvm.plan9Counter), // TODO: Temporary. Will all use a single port (9999)
-				Flags: flags,
-			},
-			ResourcePath: fmt.Sprintf("VirtualMachine/Devices/Plan9/Shares"),
-			GuestRequest: guestrequest.GuestRequest{
-				ResourceType: guestrequest.ResourceTypeMappedDirectory,
-				RequestType:  requesttype.Add,
-				Settings: guestrequest.LCOWMappedDirectory{
-					MountPath: uvmPath,
-					Port:      int32(uvm.plan9Counter), // TODO: Temporary. Will all use a single port (9999)
-					ReadOnly:  readOnly,
-				},
+	modification := &hcsschema.ModifySettingRequest{
+		RequestType: requesttype.Add,
+		Settings: hcsschema.Plan9Share{
+			Name:         name,
+			AccessName:   name,
```

Comment:
This requires a GCS change right? And its breaking to an old GCS?

Reply:
OpenGCS has supported ShareName for a while.

Reply:
Nvm. I guess we already supported the share name.